### PR TITLE
feat: allow computed properties within `useMeta`

### LINKED ISSUE #254: feat: computed properties within `useMeta`
add ability to do something like:

```ts
useMeta({
  title: 'Composition API Demo',
  meta: [{
    name: 'message',
    content: message.value // not reactive
  }]
})
```

https://github.com/nuxt-community/typescript-template/pull/225

### PR DESCRIPTION
**Pull request by danielroe**
**Usage**:
In addition to the existing usage for `useMeta`, this adds the possibility to pass a function which will effectively generate a computed head state and be merged with any other head values.
```
useMeta(() => ({ title: message.value }))
```

nuxt-community/typescript-template#225

closes #254

### CODE REVIEW

**Comment by danielroe** [comment]
@mathe42 I've reverted much of the special handling of `titleTemplate` as it seems to work now with an initial `null` value - but want to check there are no gotchas I'm unaware of. Would value your thoughts.

**Comment by mathe42** [comment]
When null doesn't overwrite the global meta data it's fine.

I can't test this in the next 2 weeks

File: src/meta.ts [hunk 1]
```diff
@@ -103,38 +103,23 @@ export const useMeta = <T extends MetaInfo>(init?: T) => {
...
-          trigger()
-        },
-      }
-    }
-  )
-
-  if (process.client)
-    watch(Object.values(refs), vm.$meta().refresh, { immediate: true })
+  if (init instanceof Function) {
+    _computedHead.push(computed(init))
```
[the remark this answers is not in the document]
Comment (danielroe) [reply src/meta.ts:113]: Have you added the `hid`property to deduplicate?

[the remark this answers is not in the document]
Comment (axelhzf) [reply src/meta.ts:113]: You are right, I am not adding the property `hid`, but I think that the property only will hide the problem.

The problem is not related to several components trying to write the same property.  I only have one component on the tree trying using `useMeta`

```js
 setup() {
    useMeta(() => {
      return {
        meta: [
          { property: 'og:title', content: 'page title' }
        ]
      }
    });
  }
```


The problem I am facing is that every request is adding a new duplicated property (I am doing server-side rendering). So I think that somehow, the state is being stored between each request.

If I print the value `_computedHead` here https://github.com/nuxt-community/composition-api/pull/255/files#diff-e0058ab35c645c7435fbc38c50432eabeec5d30d5ad15c4278e083edb897842dR106  I can see that it add a new entry on each request.


I will try to create a minimal reproduction repository.

[the remark this answers is not in the document]
Comment (axelhzf) [reply src/meta.ts:113]: Here is the repository https://github.com/axelhzf/nuxt-use-meta-bug

[the remark this answers is not in the document]
Comment (danielroe) [reply src/meta.ts:113]: Thanks @axelhzf - good catch.

**Pull request merged**